### PR TITLE
feat(authz): make v2 request limits configurable

### PR DESCRIPTION
**Pull request by jakedoublev**
Make Authorization v2 request limits configurable by moving max-count validation from proto CEL to service-side config.

## Summary by CodeRabbit

* **New Features**
  * Configurable request limits for entity chains, attribute-values, obligations, resources, and bulk decision requests; service handlers now validate and enforce limits early.

* **Documentation**
  * Updated configuration examples and extensive API/OpenAPI/gRPC docs: request-limit text, schema/validation wording, revised request/response wiring, error model shape, and various schema refactors and examples.

* **Tests**
  * Added and updated validation and handler tests covering default/custom limits, boundary cases, nested errors, and proto-validation behavior.

### CODE REVIEW

File: service/authorization/v2/validation.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		if err := as.validateGetDecisionMultiResourceRequest(decisionRequest, fmt.Sprintf("decision_requests[%d].", idx)); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (as *Service) validateEntityIdentifierRequestLimits(entityIdentifier *authzV2.EntityIdentifier, path string) error {
+	entityChain := entityIdentifier.GetEntityChain()
```
[the remark this answers is not in the document]
Comment (jakedoublev) [reply service/authorization/v2/validation.go:58]: proto-gencode getters are all nil-safe, which is why the getters exist

File: service/authorization/v2/validation.go [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+	entities := entityChain.GetEntities()
+	if len(entities) > as.config.RequestLimits.EntityChainEntitiesMax {
+		return limitExceededError(path+".entity_chain.entities", len(entities), as.config.RequestLimits.EntityChainEntitiesMax)
+	}
+
+	return nil
+}
+
+func (as *Service) validateResourceRequestLimits(resource *authzV2.Resource, path string) error {
+	attributeValues := resource.GetAttributeValues()
```
[the remark this answers is not in the document]
Comment (jakedoublev) [reply service/authorization/v2/validation.go:72]: proto-gencode getters are all nil-safe, which is why the getters exist

File: service/authorization/v2/validation.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (jakedoublev) [reply service/authorization/v2/validation.go:58]: getters exist to be nil-safe

File: service/authorization/v2/validation.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (jakedoublev) [reply service/authorization/v2/validation.go:72]: getters exist to be nil-safe

File: service/authorization/v2/config.go [hunk 3]
```diff
@@ -1,20 +1,55 @@
...
+type RequestLimitsConfig struct {
+	ResourceAttributeValuesFqnsMax              int `mapstructure:"resource_attribute_values_fqns_max" json:"resource_attribute_values_fqns_max" default:"20"`
+	EntityIdentifierEntityChainEntitiesMax      int `mapstructure:"entity_identifier_entity_chain_entities_max" json:"entity_identifier_entity_chain_entities_max" default:"10"`
+	DecisionRequestFulfillableObligationFqnsMax int `mapstructure:"decision_request_fulfillable_obligation_fqns_max" json:"decision_request_fulfillable_obligation_fqns_max" default:"50"`
+	GetDecisionMultiResourceResourcesMax        int `mapstructure:"get_decision_multi_resource_resources_max" json:"get_decision_multi_resource_resources_max" default:"1000"`
+	GetDecisionBulkDecisionRequestsMax          int `mapstructure:"get_decision_bulk_decision_requests_max" json:"get_decision_bulk_decision_requests_max" default:"200"`
+}
+
+func (c RequestLimitsConfig) Validate() error {
+	if c.ResourceAttributeValuesFqnsMax < 1 {
```
Comment (elizabethhealy) [inline service/authorization/v2/config.go:28]: should we have hard limits here as well like we did in the protos? just to ensure its not vulnerable to dos, doesnt have to be the same limit but something to avoid a footgun

Comment (jakedoublev) [reply service/authorization/v2/config.go:28]: We have sensible default limits, but I wasn't sure how it seemed to set a max configurable limit (like the max configured max limit), because we have no idea how virtually scaled the deployments could be. It didn't seem like there was an obvious big limit to set as the configurable maximum. 🤔

**Review by elizabethhealy — DISMISSED** [review]

**Review by pflynn-virtru — APPROVED** [review]

**Comment by autokyc** [comment]
/backport

Authorized security test: verifying whether a non-write external commenter can trigger the backport workflow. Existing backport PR #3656 was selected to avoid duplicate PR creation.
